packages/buckets/src/api/index.ts

```diff
@@ -80,23 +79,37 @@ export type RootObject = {
   thread: string
 }
 
+export enum PathAccessRoleMap {
+  PATH_ACCESS_ROLE_UNSPECIFIED = 0,
+  PATH_ACCESS_ROLE_READER = 1,
+  PATH_ACCESS_ROLE_WRITER = 2,
+  PATH_ACCESS_ROLE_ADMIN = 3,
+}
+
+export type Metadata = {
+  rolesMap: Array<[string, PathAccessRoleMap]>
+  updatedAt: number
+}
+
 /**
  * A bucket path item response
  */
-export type ListPathItemObject = {
+export type PathItemObject = {
   cid: string
   name: string
   path: string
   size: number
   isDir: boolean
-  itemsList: Array<ListPathItemObject>
+  itemsList: Array<PathItemObject>
+  itemsCount: number
+  metadata?: Metadata
```

Comment:
these metadata are looking a bit ugly to use the client. can we retype them to just make them easy. 

```
const convertMetadata = (metadata?: Metadata): PathMetadata | undefined => {
  if (!metadata) return
  const roles = metadata.getRolesMap()
  const typedRoles = new Map()
  roles.forEach((entry, key) => typedRoles.set(key, entry))
  const response: PathMetadata = {
    updatedAt: metadata.getUpdatedAt(),
    roles: typedRoles,
  }

  return response
}
```

Reply:
or maybe you have something nicer there. it's just the `PathAccessRoleMap` doesn't appear to act like a regular map created from scratch. you get some type mismatch :/ maybe not a huge deal. take a look though

Reply:
oop, i see. you gave that a nice type too. so maybe okay!!